src/assisted_test_infra/test_infra/helper_classes/entity.py

```diff
@@ -59,13 +64,31 @@ def prepare_for_installation(self, **kwargs):
 
         self.nodes.controller.log_configuration()
 
-        if self._config.download_image:
-            self.download_image()
+        self._config.iso_download_path = self.get_iso_download_path()
+        if self._config.download_image and not self._config.is_static_ip:
+            str(self.download_image())
+
+        self.nodes.prepare_nodes()
+        if self._config.is_static_ip:
+            # On static IP installation re-download the image after preparing nodes and setting the
```

Comment:
This is the natural order, but in the current flow we must prepare the nodes and only then set the IP configurations. To change that it will required more changes than this PR can handle :)

Reply:
I believe I'll change it somewhere in the future

Reply:
Ok